content/os/Linux Hardening Guide.md

```diff
@@ -0,0 +1,259 @@
+
+WIth that being said, if your threat model calls for using something like Kloak, you are probably better off just using Whonix.
+
+## Application Confinement
+Some sandboxing solutions for desktop Linux distributions do exist, however they are not as strict as those found in macOS or ChromeOS. Applications installed from the package manager (`dnf`, `apt`, etc.) typically have **no** sandboxing or confinement whatsoever. Below are a few projects that aim to solve this problem:
+
+### Flatpak
+
+{{< youtube id="GkgPIJp8_30">}}
+
+[Flatpak](https://flatpak.org) aims to be a universal package manager for Linux. One of its main goals is to provide a universal package format which can be used in most Linux distributions. It provides some [permission control](https://docs.flatpak.org/en/latest/sandbox-permissions.html). With that being said, Flatpak sandboxing is [quite weak](https://madaidans-insecurities.github.io/linux.html#flatpak).
+
+You can restrict applications further by issuing [Flatpak overrides](https://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpak-override). This can be done with the command-line or by using [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal). Some sample overrides are provided by [me](https://github.com/tommytran732/Flatpak-Overrides) and [rusty-snake](https://github.com/rusty-snake/kyst/tree/main/flatpak). Note that this only helps with the lax high level default permissions, but cannot solve the low level issues like `/proc` and `/sys` access, or an insufficient seccomp blacklist.
+
+Some sensitive permissions you should pay attention to:
+
+- the Network (`share=network`) socket (internet access)
+- the PulseAudio socket (for both audio in and out), `device=all` (access to all devices including the camera)
+- `org.freedesktop.secrets` dbus (access to secrets stored on your keychain) for applications which do not need it
+
+If an application works natively with Wayland (and not running through the [XWayland](https://wayland.freedesktop.org/xserver.html) compatibility layer), consider revoking its access to the X11 (`socket=x11`) and [Inter-process communications (IPC)](https://en.wikipedia.org/wiki/Unix_domain_socket) socket (`share=ipc`) as well.
+
+Many flatpak apps come with broad filesystem permissions such as `filesystem=home` and `filesystem=host`. Some applications like implement the [Portals](https://docs.flatpak.org/en/latest/portal-api-reference.html) [API](https://en.wikipedia.org/wiki/API), which allows a file manager to pass files to the Flatpak application (e.g. VLC) without specific filesystem access privileges. Despite of this, many of them, including ones like VLC [still includes](https://github.com/flathub/org.videolan.VLC/blob/master/org.videolan.VLC.json) `filesystem=host` for no apparant reason. 
```

Comment:
`filesystem=host` is somewhat questionable, but `filesystem=home:ro` is necessary to work with playlist files. For VLC specifically, being a lot more than just a media player, it's quite possible that `filesystem=home:rw` might be necessary for some features. Users that don't need those functions in VLC should certainly limit the permissions.

Reply:
filesystem=home:ro is not great. The user should only grant access to the dirs where the playlists are.

Reply:
The point isn't why VLC needs an overly permissive permission, the point is that it does not need it and a safer, more secure, and more private option of using the filesystem picker exists.

Reply:
> filesystem=home:ro is not great. The user should only grant access to the dirs where the playlists are.

I agree it is not great, but ultimately it's a tradeoff between usability and security. Most users will not think to grant VLC access to `~/Music` manually (or whatever other relevant directories); they will expect it to just work out of the box. It's much more reasonable to expect security-minded users to manually limit the permissions.

> The point isn't why VLC needs an overly permissive permission, the point is that it does not need it and a safer, more secure, and more private option of using the filesystem picker exists.

It is impossible to make playlist files work through Flatpak portals. That was my point in the first place. That is also why [Amberol (a Flatpak-only music player) does not support them](https://gitlab.gnome.org/World/amberol/-/issues/114).